website/www/site/content/en/contribute/runner-guide.md

```diff
@@ -813,83 +723,47 @@ message PCollection {
+[Overview](https://docs.google.com/presentation/d/1Cso0XP9dmj77OD9Bd53C1M3W1sPJF0ZnA20gzb2BPhE/edit#slide=id.g42e4c9aad6_1_3722)
+[Spec](https://github.com/apache/beam/blob/master/model/job-management/src/main/proto/org/apache/beam/model/job_management/v1/beam_job_api.proto)
+
+While your language's SDK will may insulate you from touching the Runner
+API protos directly, you may need to implement adapters for your runner, to
+expose it to another language.
+This allows a Python SDK to invoke a Java runner or vice versa.
+A typical implementation of this can be found in
+[local_job_service.py](https://github.com/apache/beam/blob/release-2.48.0/sdks/python/apache_beam/runners/portability/local_job_service.py)
+which is used directly to front several Python-implemented runners.
 
 The RPCs themselves will necessarily follow the existing APIs of PipelineRunner
 and PipelineResult, but altered to be the minimal backend channel, versus a
 rich and convenient API.
 
-### `PipelineRunner.run(Pipeline)` RPC
-
-This will take the same form, but `PipelineOptions` will have to be serialized
-to JSON (or a proto `Struct`) and passed along.
+A key piece of this is the
+(Artifacts API)[https://github.com/apache/beam/blob/master/model/job-management/src/main/proto/org/apache/beam/model/job_management/v1/beam_artifact_api.proto],
+which allows a Runner to fetch and deploy binary artifacts (such as jars,
+pypi packages, etc.) that are listed as dependencies in the various environments,
+and may have various representations.  This is invoked after a pipeline
+is submitted, but before it is executed. The SDK submitting a pipeline acts
+as an artifact server to the runner receiving the request, and in turn the
+runner then acts as an artifact server to the workers (environments) hosting
+the users UDFs.
 
-{{< highlight class="no-toggle" >}}
-message RunPipelineRequest {
-  Pipeline pipeline;
-  Struct pipeline_options;
-}
-{{< /highlight >}}
-
-{{< highlight class="no-toggle" >}}
-message RunPipelineResponse {
-  bytes pipeline_id;
-
-  // TODO: protocol for rejecting pipelines that cannot be executed
-  // by this runner. May just be REJECTED job state with error message.
-
-  // totally opaque to the SDK; for the shim to interpret
-  Any contents;
-}
-{{< /highlight >}}
-
-### `PipelineResult` aka "Job API"
-
-The two core pieces of functionality in this API today are getting the state of
-a job and canceling the job. It is very much likely to evolve, for example to
-be generalized to support draining a job (stop reading input and let watermarks
-go to infinity). Today, verifying our test framework benefits (but does not
-depend upon wholly) querying metrics over this channel.
-
-{{< highlight class="no-toggle" >}}
-message CancelPipelineRequest {
-  bytes pipeline_id;
-  ...
-}
-
-message GetStateRequest {
-  bytes pipeline_id;
-  ...
-}
-
-message GetStateResponse {
-  JobState state;
-  ...
-}
-
-enum JobState {
-  ...
-}
 {{< /highlight >}}
```

Comment:
This probably needs to be removed.

Reply:
Nice catch! Thanks. (Sure would be nice if it pointed us at the bad line--this brings back memories of hunting down g++ template errors...)

Fingers crossed...